### DISCUSSION TITLE
vNext "View History" UX bad?

### DISCUSSION THREAD

**Discussion by davidgiga1993** [gitextensions/gitextensions#9978] [Q&A]
In the current release version I can open a file history by double clicking or selecting "View History" from the context menu (in the file tree)
By default this opened a new window and selected the last commit automatically, immediatly showing the last change made in that particular file.

In vNext the same steps result in a new window, but no commit is selected, resulting in an nearly completely empty window.

Is this the intended behavior?
![image](https://user-images.githubusercontent.com/3924958/165045135-a031adbf-b0d7-4310-9183-f1d52c2bfc37.png)

# Improvement Ideas
- It would be much nicer if at least the latest commit would be automatically selected.
- The "file history" window should remember the last tab I've selected (for example "Diff" or "File Tree")
- Compared to the old way, the new window takes much longer to open (guess since it's a separate process instance now?). I'm sure there was some requirement that resulted in this solution, however it would be nice to have faster startup of this.

**Comment by pmiossec** [comment]
I don't reproduce...

Is it possible that you set a filter that prevent the good commit to be displayed? In this filter settings popup:
![image](https://user-images.githubusercontent.com/460196/165057505-fc7172ad-312b-4e80-b020-645a6654ba70.png)

**Reply by davidgiga1993** [reply]
No there is no filter set.. It happens when you open the history of a file (via File tree, **not** via diff). Using git extensions `4.0.0.13768`

**Reply by crypto-rsa** [reply]
Does the file name include a space? I've encountered issues with such files, though I haven't logged them yet 😞

**Reply by davidgiga1993** [reply]
No, but it happens every time if the file has only one commit

**Reply by gerhardol** [reply]
You can get the old UI back in Settings-Detailed-Browse
(There is no configuration to show the Diff tab instead of FileTree with Blame, but if you filter a WorkTree file Diff is shown as FileTree is not visible).
The load  times is very similar for the old and the new UI though. (In general, the load times for loading revisions is much optimized in vNext.)
You can also filter files in the Browse UI without opening a new instance.

If this was about selections of commits I would handle selection or not separate from UI discussions. That is similar in the the two forms.
That is not much changed to 3.5 (some changes in #9954 though).

Personally, I would prefer to have Diff tab default in all situations.

**Reply by davidgiga1993** [reply]
Thanks I was not aware of that setting, that helps :)

Yes I usually don't filter in the browse UI, since the main window usually focusses on the commits itself the height of the diff/file content view is much smaller. That's why I preferr a separate window since its layout can be resized to show all the file content and only 1-2 commits.

The issue I was mentioning above is that sometimes when opening the file window none of the commits are selected, making it mandatory to manually click on the commit. So I guess this might be a bug?
Also when using the old UI it's also blank sometimes when double clicking a file from the file explorer tab:
![image](https://user-images.githubusercontent.com/3924958/165242196-4d536c33-7372-4699-9798-4b183fcd109f.png)

This seems to happen if a file has only been added (only one commit touched the file)
